server/service/client.go

```diff
@@ -1535,30 +1674,85 @@ func (c *Client) DoGitOps(
 	return teamAssumptions, nil
 }
 
-func (c *Client) doGitOpsNoTeamSoftware(config *spec.GitOps, baseDir string, appconfig *fleet.EnrichedAppConfig, logFn func(format string, args ...interface{}), dryRun bool) ([]fleet.SoftwarePackageResponse, error) {
+func (c *Client) doGitOpsNoTeamSoftware(
+	config *spec.GitOps,
+	baseDir string,
+	appconfig *fleet.EnrichedAppConfig,
+	logFn func(format string, args ...interface{}),
+	dryRun bool,
+) ([]fleet.SoftwarePackageResponse, error) {
+	if !config.IsNoTeam() || appconfig == nil || !appconfig.License.IsPremium() {
+		return nil, nil
+	}
+
 	var softwareInstallers []fleet.SoftwarePackageResponse
-	if config.IsNoTeam() && appconfig != nil && appconfig.License.IsPremium() {
-		packages := make([]fleet.SoftwarePackageSpec, 0, len(config.Software.Packages))
-		for _, software := range config.Software.Packages {
-			if software != nil {
-				packages = append(packages, *software)
+
+	packages := make([]fleet.SoftwarePackageSpec, 0, len(config.Software.Packages))
+	packagesByPath := make(map[string]fleet.SoftwarePackageSpec, len(config.Software.Packages))
+	for _, software := range config.Software.Packages {
+		if software != nil {
+			packages = append(packages, *software)
+			if software.ReferencedYamlPath != "" {
+				// can be referenced by macos_setup.software
+				packagesByPath[software.ReferencedYamlPath] = *software
 			}
 		}
-		payload, err := buildSoftwarePackagesPayload(baseDir, packages)
-		if err != nil {
-			return nil, fmt.Errorf("applying software installers: %w", err)
-		}
-		logFn("[+] applying %d software packages for 'No team'\n", len(payload))
-		softwareInstallers, err = c.ApplyNoTeamSoftwareInstallers(payload, fleet.ApplySpecOptions{DryRun: dryRun})
+	}
+
+	// marshaling dance to get the macos_setup data
```

Comment:
Nit: Needs better comment here. This is to get additional validation and get optjson functionality, right?

Reply:
It's because we have it as a `map[string]interface{}` stored in a `interface{}` here, and hand-unwrapping all this is not super robust or easy to maintain. I'll update the comment to make that clearer.